include/deal.II/hp/dof_handler.h

```diff
@@ -1213,33 +1213,45 @@ namespace hp
     std::vector<unsigned int> vertex_dof_offsets;
 
     /**
-     * Container to temporarily store the iterator and active FE index of
-     * cells that will be refined.
+     * Whenever the underlying triangulation changes by either
+     * refinement/coarsening and serialization, the active_fe_index of cells
+     * needs to be transferred. This structure stores all temporary information
+     * required during that process.
      */
-    std::map<const cell_iterator, const unsigned int> refined_cells_fe_index;
-
-    /**
-     * Container to temporarily store the iterator and active FE index of
-     * parent cells that will remain after coarsening.
-     */
-    std::map<const cell_iterator, const unsigned int> coarsened_cells_fe_index;
-
-    /**
-     * Container to temporarily store the active_fe_index of every locally
-     * owned cell for transfer across parallel::distributed::Triangulation
-     * objects.
-     */
-    std::vector<unsigned int> active_fe_indices;
-
-    /**
-     * Helper object to transfer all active_fe_indices on
-     * parallel::distributed::Triangulation objects during refinement/coarsening
-     * and serialization.
-     */
-    std::unique_ptr<
-      parallel::distributed::
-        CellDataTransfer<dim, spacedim, std::vector<unsigned int>>>
-      cell_data_transfer;
+    struct ActiveFEIndexTransfer
+    {
+      /**
+       * Container to temporarily store the iterator and active FE index of
+       * cells that will be refined.
+       */
+      std::map<const cell_iterator, const unsigned int> refined_cells_fe_index;
+
+      /**
+       * Container to temporarily store the iterator and active FE index of
+       * parent cells that will remain after coarsening.
+       */
+      std::map<const cell_iterator, const unsigned int>
+        coarsened_cells_fe_index;
+
+      /**
+       * Container to temporarily store the active_fe_index of every locally
+       * owned cell for transfer across parallel::distributed::Triangulation
+       * objects.
+       */
+      std::vector<unsigned int> active_fe_indices;
+
+      /**
+       * Helper object to transfer all active_fe_indices on
+       * parallel::distributed::Triangulation objects during
+       * refinement/coarsening and serialization.
+       */
+      std::unique_ptr<
+        parallel::distributed::
+          CellDataTransfer<dim, spacedim, std::vector<unsigned int>>>
+        cell_data_transfer;
+    };
+
+    ActiveFEIndexTransfer active_fe_index_transfer;
```

Comment:
You'll need a line of documentation :-)

Since this object only lives during a particular time in the lifetime of the `hp::DoFHandler`, I had thought that maybe one could make it a `std::unique_ptr`. This way, it is clear when the object is live: when the pointer is non-`nullptr`. In particular, one can create the transfer object before filling it, and destroy it after.